packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -2501,5 +2507,309 @@ describe("Runtime", () => {
+				// Missing signal should be detected
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 1,
+						},
+					],
+					"SignalLost telemetry should be logged when signal is dropped",
+				);
+
+				dropSignals(2);
+
+				processSubmittedSignals(1);
+
+				// Missing 3rd and 4th signals should be detected
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 2,
+						},
+					],
+					"SignalLost telemetry should be logged when signal is dropped",
+				);
+			});
+
+			it("ignores in-flight signals on disconnect/reconnect", () => {
+				sendSignals(4);
+
+				// Disconnect + Reconnect
+				changeConnectionState(containerRuntime, false, mockClientId);
+				changeConnectionState(containerRuntime, true, mockClientId);
+
+				// Temporarily lose two old signals
+				dropSignals(2);
+
+				// Receive old signals sent before disconnect
+				processSubmittedSignals(2);
+
+				// Receive old out of order signals
+				processDroppedSignals(2);
+
+				// No error events should be logged for signals sent before disconnect
+				logger.assertMatchNone(
+					[
+						{
+							eventName: "ContainerRuntime:SignalOutOfOrder",
+						},
+						{
+							eventName: "ContainerRuntime:SignalLost",
+						},
+					],
+					"SignalOutOfOrder/SignalLost telemetry should not be logged on reconnect",
+				);
+			});
+
+			it("counts both relative and abosolute lost signal counts", () => {
+				sendSignals(60);
+
+				processSubmittedSignals(10);
+
+				dropSignals(1);
+
+				processSubmittedSignals(39);
+
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 1,
+						},
+					],
+					"SignalLost telemetry should log relative lost signal count when a signal is dropped",
+				);
+
+				dropSignals(5);
+
+				sendSignals(45);
+
+				processSubmittedSignals(30);
+
+				dropSignals(4);
+
+				// Process remaining signals
+				processSubmittedSignals(16);
+
+				logger.assertMatch(
+					[
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 5,
+						},
+						{
+							eventName: "ContainerRuntime:SignalLost",
+							signalsLost: 4,
+						},
+						{
+							eventName: "ContainerRuntime:SignalLatency",
+							signalsLost: 10,
+						},
+					],
+					"SignalLost telemetry should log relative lost signal count and SignalLatency telemetry should log absolute lost signal count for each batch of 100 signals",
+				);
+			});
+
+			it("accurately reports amount of sent and lost signals when roundtrip tracked signal is droppped", () => {
+				// Send 50 signals and drop 10
+				sendSignals(50);
+				dropSignals(10);
+				processSubmittedSignals(40);
+
+				// Send 60 signals and drop 10 (including roundtrip tracked signal)
+				sendSignals(60);
+				processSubmittedSignals(40);
+				// Drop roundtrip tracked signal
+				dropSignals(10);
+				processSubmittedSignals(10);
```

Comment:
If dropped was 5 higher (15) and processed was 5 lower (5), shouldn't the lost count go up in the next Latency report?
I tried that and it doesn't. I am not sure how we really make full sense of things.

If I just change the processed to
				processSubmittedSignals(5);
				dropSignals(5);
then the lost count is 6.

Reply:
I thought I should double check that I didn't have the right expectation for counts when I tried that first varying sequence. The result was more confusing. This passes:
```typescript
				// Send 50 signals and drop 10
				sendSignals(50);
				dropSignals(10);
				processSubmittedSignals(40);

				// Send 60 signals and drop 10 (including roundtrip tracked signal)
				sendSignals(60);
				processSubmittedSignals(40);
				// Drop roundtrip tracked signal
				dropSignals(15);
				processSubmittedSignals(5);

				// Send 100 signals and drop 1
				sendSignals(100);
				dropSignals(1);
				processSubmittedSignals(99);

				// Check SignalLatency only logs the amount of lost signals in latest 100 signal batch
				logger.assertMatch(
					[
						{
							eventName: "ContainerRuntime:SignalLatency",
							signalsLost: 1,
							signalsSent: 200,
						},
					],
					"SignalLatency telemetry should log absolute lost signal count for each batch of 100 signals",
				);
```
clearly we lost more than 1 of the 200.

Reply:
I mistakenly would reset the signalsLost when the roundtrip signal was detected as lost. You should now see the 26 lost signals corresponding with the 200 sent.

Reply:
Re 1st message: In this case I was testing for the 2nd SignalLatency event with 1:100 lost to sent ratio. I can now see this wasn't very clear since I didn't explicitly test for the first 20:100 event and the comments were just wrong (the roundtrip signal isn't actually lost).